openmc/mgxs/mgxs.py

```diff
@@ -1484,22 +1487,26 @@ def rxn_rate_tally(self):
             self._rxn_rate_tally.sparse = self.sparse
         return self._rxn_rate_tally
 
+class FissionXSBase(MGXS):
+    """A fission production multi-group cross section base class
+       for NuFission and KappaFission
+    """
 
-class FissionXS(MGXS):
-    """A fission multi-group cross section."""
+    # This is an abstract class which cannot be instantiated
+    __metaclass__ = abc.ABCMeta
```

Comment:
Agreed that using Python opens things up for misuse. But what I proposed above doesn't seem to add much code - it would replace 4 lines of code which assign the `_rxn_type` class attribute with 8 lines of code for the properties. In addition, this would prevent a user from accidentally over-riding the `_rxn_type` attribute within the class. I personally find this more compelling than option 1). Option 2) may make more sense in the future as more `MGXS` subclasses are added and we can see how best to generalize this into a factor pattern.